### PR TITLE
Version2: Add certificate search logic

### PR DESCRIPTION
**Pull request by OleksandrSPV**
https://github.com/PeculiarVentures/fortify-tools/assets/129727832/f4ce0374-2b5e-4a95-b82b-aaf9e3612bcc

### CODE REVIEW

File: src/hooks/search-list/useSearchList.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from "react";
+import { CertificateProps } from "../../types";
+
+export function useSearchList(certificates: CertificateProps[]) {
+  const searchParams = new URLSearchParams(window.location.search);
```
Comment (donskov) [inline src/hooks/search-list/useSearchList.tsx:5]: We need to use the `searchParams` only for default state prefilling, but this code calls on each rerender. We have to fix this and parse parameters only 1 time.

File: src/hooks/search-list/useSearchList.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    return () => {
+      window.removeEventListener("popstate", handleUrlChange);
+    };
+  }, []);
+
+  return {
+    searchedText,
+    list: searchedText
+      ? certificates.filter(({ label }) =>
```
Comment (donskov) [inline src/hooks/search-list/useSearchList.tsx:40]: It's the same situation with multiple re-renders. We should try to make filtration only when the `searchedText` or the `certificates` have changed. Now, even when opening a dialog, we filter certificates.

**Review by donskov — CHANGES_REQUESTED** [review]

**Comment by donskov** [comment]
@OleksandrSPV Not related to these changes, but is there any reason to have the search state in the search hook and two search states in the [component](https://github.com/PeculiarVentures/fortify-tools/blob/donskov/v2/src/components/certificates-topbar/CertificatesTopbar.tsx#L34-L35)? Can we use only a single source of truth?

File: src/hooks/search-list/useSearchList.tsx (hunk 1 above)
Comment (OleksandrSPV) [reply src/hooks/search-list/useSearchList.tsx]: @donskov Fixed here https://github.com/PeculiarVentures/fortify-tools/pull/167/commits/3de8b750b325d4703ad0f8c94cf521823f848876

File: src/hooks/search-list/useSearchList.tsx (hunk 2 above)
Comment (OleksandrSPV) [reply src/hooks/search-list/useSearchList.tsx]: @donskov Fixed here https://github.com/PeculiarVentures/fortify-tools/pull/167/commits/3de8b750b325d4703ad0f8c94cf521823f848876

**Comment by OleksandrSPV** [comment]
> @OleksandrSPV Not related to these changes, but is there any reason to have the search state in the search hook and two search states in the [component](https://github.com/PeculiarVentures/fortify-tools/blob/donskov/v2/src/components/certificates-topbar/CertificatesTopbar.tsx#L34-L35)? Can we use only a single source of truth?

@donskov I needed to use delayed input, so I decided use this part near the input component not in search hook. If you have some proposition, please let me know.

**Comment by donskov** [comment]
> > @OleksandrSPV Not related to these changes, but is there any reason to have the search state in the search hook and two search states in the [component](https://github.com/PeculiarVentures/fortify-tools/blob/donskov/v2/src/components/certificates-topbar/CertificatesTopbar.tsx#L34-L35)? Can we use only a single source of truth?
> 
> @donskov I needed to use delayed input, so I decided use this part near the input component not in search hook. As I see guys on Goodkey don't use input delay for certificate search, we can go this way if you prefer, and remove one state from search component, or if you have another proposition, please let me know.

Yeah, let's remove it.

**Comment by OleksandrSPV** [comment]
> > > @OleksandrSPV Not related to these changes, but is there any reason to have the search state in the search hook and two search states in the [component](https://github.com/PeculiarVentures/fortify-tools/blob/donskov/v2/src/components/certificates-topbar/CertificatesTopbar.tsx#L34-L35)? Can we use only a single source of truth?
> > 
> > 
> > @donskov I needed to use delayed input, so I decided use this part near the input component not in search hook. As I see guys on Goodkey don't use input delay for certificate search, we can go this way if you prefer, and remove one state from search component, or if you have another proposition, please let me know.
> 
> Yeah, let's remove it.

@donskov removed here https://github.com/PeculiarVentures/fortify-tools/pull/167/commits/93c9ee78a122b89aa091333903ab4a75aa9ec284

**Comment by donskov** [comment]
> > > > @OleksandrSPV Not related to these changes, but is there any reason to have the search state in the search hook and two search states in the [component](https://github.com/PeculiarVentures/fortify-tools/blob/donskov/v2/src/components/certificates-topbar/CertificatesTopbar.tsx#L34-L35)? Can we use only a single source of truth?
> > > 
> > > 
> > > @donskov I needed to use delayed input, so I decided use this part near the input component not in search hook. As I see guys on Goodkey don't use input delay for certificate search, we can go this way if you prefer, and remove one state from search component, or if you have another proposition, please let me know.
> > 
> > 
> > Yeah, let's remove it.
> 
> @donskov removed here [93c9ee7](https://github.com/PeculiarVentures/fortify-tools/commit/93c9ee78a122b89aa091333903ab4a75aa9ec284)

Do we also need the search input state (`const [searchInputValue, setSearchInputValue] = React.useState(searchValue);`)?

**Comment by OleksandrSPV** [comment]
> 235b1fd

@donskov oops, actually don't, fixed here

**Review by donskov — APPROVED** [review]

**Pull request merged**